eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -0,0 +1,54 @@
+{
+  "displayNames": {
+    "/p:UseProjectReferenceToAzureClients=true": "UseProjectRef",
+    "true": "Record"
+  },
+  "matrix": {
+    "Agent": {
+      "windows-2019": {
+        "OSVmImage": "MMS2019",
+        "Pool": "azsdk-pool-mms-win-2019-general"
+      }
+    },
+    "TestTargetFramework": [
+      "netcoreapp2.1",
+      "net461",
+      "net5.0"
+    ],
+    "AdditionalTestArguments": [
+      "",
+      "/p:UseProjectReferenceToAzureClients=true"
+    ]
+  },
+  "include": [
+    {
+      "Agent": {
+        "ubuntu-18.04": {
```

Comment:
Why did you do each agent as a separate include?

Reply:
It depends on whether we want to keep the exact test matrix that exists in the repo or not. As it is in master, the matrix runs every framework against windows, with both project references included and not included. If we're open to changing the matrix, then I can move linux out of the includes and into the base matrix, but I wasn't sure what the reasoning behind all the windows configurations was.

Reply:
net461 is the only supported on windows as well as UseProjectReferenceToAzureClients so those 2 should be pulled out into an include but I think the other matrix entries should include all the OS agents and doing that sparse I believe will be fine.

Reply:
Ah ok, that makes sense. I'll re-arrange to the same effect.

Reply:
@weshaggard I swapped it around, but it's still got a few extra includes, in order to support the record mode, and limit the macos agent only to public cloud.